### PR TITLE
gl_engine: revise the gl portability

### PR DESCRIPTION
**Pull request by hermet**
- guarantee minimum version requirement
- removed glesv2 dependency
- corrected gl ver dependency
- removed sdl practices

issue: https://github.com/thorvg/thorvg/issues/2282

### CODE REVIEW

File: src/renderer/gl_engine/tvgGlCommon.h [hunk 1]
```diff
@@ -26,12 +26,17 @@
...
     #include <GLES3/gl3.h>
+    #define TVG_REQUIRE_GL_MAJOR_VER 3
+    #define TVG_REQUIRE_GL_MINOR_VER 0
 #elif defined (THORVG_GL_TARGET_GL)
     #if defined(__APPLE__) || defined(__MACH__)
         #include <OpenGL/gl3.h>
     #else
-        #include <GL/gl3.h>
+        #define GL_GLEXT_PROTOTYPES 1
+        #include <GL/gl.h>
```
Comment (hermet) [inline src/renderer/gl_engine/tvgGlCommon.h:36]: @RuiwenTang I couldn't find `<GL/gl3.h>` please confirm this, Thanks.

File: examples/Example.h [hunk 2]
```diff
@@ -317,19 +317,7 @@ struct GlWindow : Window
...
-
-        if (window == nullptr) {
-            // try create OpenGLES context failed, create OpenGL context instead
-            SDL_GL_SetAttribute(SDL_GL_CONTEXT_MAJOR_VERSION, 3);
-            SDL_GL_SetAttribute(SDL_GL_CONTEXT_MINOR_VERSION, 3);
-            SDL_GL_SetAttribute(SDL_GL_CONTEXT_PROFILE_MASK, SDL_GL_CONTEXT_PROFILE_CORE);
-            window = SDL_CreateWindow("ThorVG Example (OpenGL)", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, width, height, SDL_WINDOW_OPENGL | SDL_WINDOW_HIDDEN | SDL_WINDOW_RESIZABLE);
-        }
-
+        window = SDL_CreateWindow("ThorVG Example (OpenGL/ES)", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED, width, height, SDL_WINDOW_OPENGL | SDL_WINDOW_HIDDEN | SDL_WINDOW_RESIZABLE);
```
Comment (hermet) [inline examples/Example.h:320]: @RuiwenTang Please let me know if this brings up an issue for you. Thanks.

File: src/renderer/gl_engine/tvgGlCommon.h (hunk 1 above)
Comment (RuiwenTang) [reply src/renderer/gl_engine/tvgGlCommon.h:36]: which platform do you compile to ?

File: examples/Example.h (hunk 2 above)
Comment (RuiwenTang) [reply examples/Example.h:320]: If not specify GL context version, on MacOS, SDL will create a GL 2.0 context.

File: src/renderer/gl_engine/tvgGlCommon.h (hunk 1 above)
Comment (hermet) [reply src/renderer/gl_engine/tvgGlCommon.h:36]: linux(ubuntu)

Comment (RuiwenTang) [reply src/renderer/gl_engine/tvgGlCommon.h:36]: I will check this on my Ubuntu environment.

BTW:
Maybe we should consider dynamically loading GL functions, which would avoid most compilation and linking issues

File: examples/Example.h (hunk 2 above)
Comment (hermet) [reply examples/Example.h:325]: @RuiwenTang Please double-check it again. The problem might be with the profile (OpenGL ES), not the version. In my case, it's 4.1 by default. I believe this should work.

Comment (RuiwenTang) [reply examples/Example.h:325]: `4.1` means target on OpenGL context, highest version in GLES is 3.2. And I think the `SDL_GL_CONTEXT_PROFILE_CORE` only valid when target on OpenGL.

Comment (hermet) [reply examples/Example.h:325]: @RuiwenTang the main point is it. SDL would use the higher version based on the designated profile. And gl engine should work on it.

Comment (RuiwenTang) [reply examples/Example.h:329]: yes 330 core is the minimum version GLEngine required. It works in higher version

**Pull request merged**